src/librustc_mir/build/mod.rs

```diff
@@ -341,6 +341,11 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
         }
     }
+
+    pub fn check_overflow(&self) -> bool {
```

Comment:
Some discussion on IRC between @alexcrichton and I is [here](https://botbot.me/mozilla/rustc/2016-05-03/?msg=65378260&page=2). The conclusion was:

```
14:26 <•acrichto> nmatsakis: so if libcore is implicitly relying on overflow checks in monomorphizations, that's bad
14:26 <•acrichto> e.g. something that should be fixed in libcore with explicit checks
14:27 <•nmatsakis> yeah I was wondering about that
14:27 <•acrichto> otherwise I'd be fine just r+'ing the patch and seeing how bors reacts
14:27 <•nmatsakis> whether this was the sol'n
14:27 <•nmatsakis> it seems like libcore ought to be clear about what it provides
14:27 <•nmatsakis> and if we really want it to "behave as the downstream crate behaves"
14:27 <•nmatsakis> we could add some intrinsics for that effect
14:27 <•nmatsakis> er, behave as the point of instantiation behaves,
14:27 <•nmatsakis> but I doubt that this would ever be waht we want
14:27 <•nmatsakis> since it seems inherently... unspecified.
14:27 <•acrichto> nmatsakis: yeah nah we should always be explicit
14:28 <•acrichto> it's somewhat of a bug that compiling core in release mode still monomorphizes overflow checks
```